x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_jobs.tsx

```diff
@@ -87,15 +94,39 @@ export const useLogAnalysisJobs = ({
     [spaceId, sourceId, timeField, bucketSpan]
   );
 
+  const indexPatterns = indexPattern.split(',');
+
+  const [availableIndices, setAvailableIndices] = useState<AvailableIndex[]>(
+    indexPatterns.map<AvailableIndex>(pattern => ({ indexPattern: pattern, valid: false }))
+  );
+
   const [fetchJobStatusRequest, fetchJobStatus] = useTrackedPromise(
     {
       cancelPreviousOn: 'resolution',
       createPromise: async () => {
         dispatch({ type: 'fetchingJobStatuses' });
-        return await callJobsSummaryAPI(spaceId, sourceId);
+        return Promise.all([
+          callJobsSummaryAPI(spaceId, sourceId),
+          callIndexPatternsValidate({
```

Comment:
Should this really be called whenever we fetch the job status? This happens every 30 seconds on the results page, where the validation result is not used.

Might be sufficient to only call it when the setup form changes. And then would it make sense to move it into the `useAnalysisSetupState` hook, where the `selectedIndices` are managed and `validationErrors` are already derived?

Reply:
Oh, I didn't realise we were polling this.

I think it makes sense to run the validation only once when the page loads, and use the cache response to show/hide the errors when the user selects/deselects the indices that they want for the jobs.

The UI feels snappier like that (the user doesn't need to wait for the validation call to finish). Maybe I can move the call to its own `useTrackedPromise` or combine it with something else.

Reply:
That sounds reasonable, a separate tracked promise (inside of `useAnalysisSetupState`) makes most sense to me :+1:

I wonder how we want the validation to interact with the form. Should the invalid indices be represented by disabled checkboxes? If so, how do we tell the user the reason?

Or should we just display a validation error once the user checks the box for an invalid index?

Reply:
> Or should we just display a validation error once the user checks the box for an invalid index?

I'm doing this now.

> Should the invalid indices be represented by disabled checkboxes? If so, how do we tell the user the reason?

Another option is to remove the checkboxes altogether. We can add a small note on the lines of "⚠️ Some of your indices cannot be analysed", and show a popover with the error messages if the user hovers over it.

Reply:
I think we should keep the checkboxes for so the user can easily see what is available and what is not.

Reply:
I've changed the call point for the API. I think it's better now.

I've not disabled yet the checkboxes. I cannot think of a way to show why they are disabled in a user friendly way (✈️😴). What I have done is to not check by default the indices that have errors. If the user decides to select them we still disable the submit button and show an error.